### ISSUE TITLE
libwx_gtk3u_html-3.0 called but missing on service regisister

### ISSUE THREAD

**Issue by Inmused-Dev** [federico-terzi/espanso#2096]
### Description

I have been trying to install on debian bookworm (fresh install, latest stable)
when running `espanso service register` i get the following error

 `espanso: error while loading shared libraries: libwx_gtk3u_html-3.0.so.0: cannot open shared object file: No such file or directory`

From my noob research libwx_gtk3u_html-3.0 has been updated to libwx_gtk3u_html-3.2.   

### Steps to reproduce

1. Followed the install procedure - failed
2. Installed libwxgtk3.2-dev and reran the install procedure and it still failed as it calls on 3.0 

### Expected behavior

Service should register

### Your environment

- Public source code:
- Environment name and version (e.g., Rust 1.59.0):
- Operating system and version (e.g., Ubuntu 20.04.2 LTS):
- Espanso version (e.g., 2.1.4-beta):

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by KatayR** [comment]
Same

**Comment by pnunn** [comment]
Just upgrade my NEON install to the latest version (24.04 base) and now have the same error with version  2.2.1)

**Comment by Inmused-Dev** [comment]
Seems that if this bug is causing install failures, they it should be a top priority?

**Comment by TheNomad11** [comment]
After updating to Linux Mint 22 I have the same issue, others as well https://forums.linuxmint.com/viewtopic.php?t=422703

**Comment by petr-fischer** [comment]
This error occurs on my system after upgrading from Ubuntu 24.04 to the 24.10.

**Comment by dddw** [comment]
can confirm having this issue on 24.10 on wayland via .deb install.
can't we just get a flatpak ?

**Comment by petr-fischer** [comment]
> can confirm having this issue on 24.10 on wayland via .deb install. can't we just get a flatpak ?

I'm not sure flatpak (or snap) is suitable (easy to use, even though it solves dependencies) for such things plugged "into" the system (I also noticed that when espanso was running, I had some annoying fake window in the dock with minimum size, maybe it catches keys through it, it's kinda hacky).
A repository with package updates would suffice :)
/* Maybe we should donate/donate more */

**Comment by dddw** [comment]
> /* Maybe we should donate/donate more */

you are right, will add it to my  open-source/software-donation-rotation since this is really the first crossplatform text expander, and I really like it.

**Comment by fewaltix** [comment]
Similar issue here - on OpenSuse, it complains that "libwx_gtk3u_html-suse.so.11.0.0" is missing, after the last distribution upgrade updated this package to "libwx_gtk3u_html-suse12_0_0".

**Comment by hunterjackson** [comment]
Happening on Fedora 40

**Comment by pnunn** [comment]
Just discovered that espanso service is constantly crashing on my system because of this, but I can't find any way to stop it. I try to run espanso anything and it throws the library error, but, I can't find the service in systemd so I can't work out how to stop it restarting.

**Comment by karlkatzke** [comment]
Also having this problem on 2.2.1 with Debian 24.04 LTS.

**Comment by fuguesoft** [comment]
Just made an issue about this before seeing this one:
https://github.com/espanso/espanso/issues/2192

**Comment by leggewie** [comment]
> Also having this problem on 2.2.1 with Debian 24.04 LTS.

same here on Ubuntu 24.04 LTS Noble

**Comment by smeech** [comment]
Duplicate of #1793, #1530, #2192 (closed), #1986 (closed).

Solutions:
- compile from source
- [add `libwx_gtk3u_html-3.0.so.0`](https://github.com/espanso/espanso/issues/1793#issuecomment-1919055522)

It'll be fixed in the next release, once we can sort out the certification issues.

**Comment by fuguesoft** [comment]
I haven't been able to compile from source either
I listed it in my issue (#2192) but updated it to make that more clear

**Comment by habi** [comment]
Same for me on a fresh install of Linux Mint

````
❯ cat /etc/os-release
NAME="Linux Mint"
VERSION="22 (Wilma)"
ID=linuxmint
ID_LIKE="ubuntu debian"
PRETTY_NAME="Linux Mint 22"
VERSION_ID="22"
[...]
VERSION_CODENAME=wilma
UBUNTU_CODENAME=noble
````

After following the steps on https://espanso.org/docs/install/linux/, I get 
````
❯ espanso
espanso: error while loading shared libraries: libwx_gtk3u_html-3.0.so.0: cannot open shared object file: No such file or directory
````
`sudo apt-get install libwxgtk3.2-dev` doesn't help, I get the same error message.

**Comment by fuguesoft** [comment]
Just wanted to contribute something I learned recently concerning the current Debian 12 workaround listed in #1793 

The aforementioned workaround is something that is not recommended per debian's official docs: https://wiki.debian.org/DontBreakDebian

Namely, mixing repositories from different distros (i.e. doing a "FrankenDebian") can break your install in subtle ways that won't be noticed until much later down the road.

I don't know if the workaround could be updated with a disclaimer for the time being or if it's even that high priority to warrant it

Thank you for your time  🙏🏾 🙏🏾 🙏🏾 🙏🏾

**Comment by dstark** [comment]
> Namely, mixing repositories from different distros (i.e. doing a "FrankenDebian") can break your install in subtle ways that won't be noticed until much later down the road.

@FugueSoft, thanks for this suggestion, and @davidvv, thanks so much for proposing the workaround you have, and kudos for adding the disclaimer. I've tried the procedure twice on Debian 12 with `deb http://mirrors.kernel.org/ubuntu focal main universe`, and the OS broke badly both times (thankfully on fresh installations but once before I had Timeshift installed).

I haven't entirely traced whether it breaks simply on doing `sudo apt update && sudo apt upgrade` or whether the breakdown happens only after trying to install `libwx_gtk3u_html-3.0`. When installing `libwx_gtk3u_html-3.0`, a couple dependencies remain unresolved and a couple upgrades get held back, but nothing initially seems to break. After restarting, however, boot time is significantly longer, the power management window only loads with an error saying the service isn't running, system apps take much longer to load, applications sometimes open in the background rather than the foreground, etc.

Grateful to hear (@smeech) that a fix is upcoming in the next release. I'll sit tight and try again after that point. But just wanted to add the first-hand account of things going badly wrong at least for me with what otherwise seemed to be a comparatively straightforward workaround. So, this isn't a purely theoretical problem.

**Comment by elisspace** [comment]
Same issue on Debian 12. Looks like something is coming soon so I'll keep an eye on this thread. Thanks @smeech !

**Comment by felipoantonoff** [comment]
Here we have the same problem with Debian 12.9 with XFCE, installed using the official Espanso .deb.

Libwxgtk3.0-gtk3-dev is an obsolete package in Debian 12 and I believe that other distributions now use libwxgtk3.2-dev, which is installed normally with:
sudo apt install libwxgtk3.2-dev

The problem is that even using libwxgtk3.2-dev, the error occurs:
espanso: error while loading shared libraries: libwx_gtk3u_html-3.0.so.0: cannot open shared object file: No such file or directory

Wouldn't it be easier for Espanso to update to use version 3.2 or newer? From what I've seen, this problem has occurred with other people for some time now.

I saw the option to change the sources.list, but I think it would be better to make the correction in the Espanso code.

I think it is not recommended to use an old lib that Debian 12 itself has already updated, in fact FugueSoft even mentioned the link about this problem https://wiki.debian.org/DontBreakDebian and in other topics about the same problem it was unfortunately closed without a real solution.

**Comment by sphh** [comment]
I might be wrong, but the problem might be here: #2126 (and #1928)

**Comment by AucaCoyan** [comment]
Hey everyone!
We have a recent release. I hope the new one solves everything 😊. 
This was also reported in #1793 , with fix instructions.
If anybody is having problems, please comment in [the discord](https://discord.gg/DFcCNDg7bB) or here and we follow up the conversation.
Have a nice day!

**Comment by cosmic-hobo** [comment]
hi there, unfortuately the problem persists for me also with the new release... i followed the ofificial installation instruction. debian 12 on x11. i ran into the issue with the older release and thought i'm waiting for the new one....

$ espanso status
espanso: error while loading shared libraries: libwx_gtk3u_html-3.0.so.0: cannot open shared object file: No such file or directory
$ espanso start
espanso: error while loading shared libraries: libwx_gtk3u_html-3.0.so.0: cannot open shared object file: No such file or directory

**Comment by AucaCoyan** [comment]
Mmmm, this a possible is a missconfiguration of your system 🤔 . It works on various other machines.
Let us know in the [espanso discord](https://discord.gg/DFcCNDg7bB)

**Comment by freebrowser1** [comment]
Same issue here, Linux Mint / Ubuntu 24.04 using XFCE4 and gdm.

`Linux my-mint 6.8.0-60-generic #63-Ubuntu SMP PREEMPT_DYNAMIC Tue Apr 15 19:04:15 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux`
